crates/primitives/src/signature/super_signature.rs

```diff
@@ -0,0 +1,30 @@
+use crate::{Parity, U256};
+
+#[cfg(not(feature = "rlp"))]
+pub trait RlpSuperSig {}
+
+#[cfg(feature = "rlp")]
+pub trait RlpSuperSig: alloy_rlp::Encodable + alloy_rlp::Decodable {}
+
+#[cfg(not(feature = "k256"))]
+pub trait K256SuperSig {}
+
+#[cfg(feature = "k256")]
+pub trait K256SuperSig: From<(k256::ecdsa::Signature, k256::ecdsa::RecoveryId)> {}
+
+#[cfg(not(feature = "serde"))]
+pub trait SerdeSuperSig {}
+
+#[cfg(feature = "serde")]
+pub trait SerdeSuperSig: serde::Serialize + serde::Deserialize {}
+
+#[cfg(not(feature = "arbitrary"))]
+pub trait ArbitrarySuperSig {}
+
+#[cfg(feature = "arbitrary")]
+pub trait ArbitrarySuperSig:
+arbitrary::Arbitrary
++ proptest::arbitrary::Arbitrary<Parameters=(),Strategy=proptest::strategy::FilterMap<
+    <(U256, U256, Parity) as proptest::arbitrary::Arbitrary>::Strategy,
+    fn((U256, U256, Parity)) -> Option<Self>,
+>> {}
```

Comment:
should be enough to do
```suggestion
pub trait ArbitrarySuperSig:
for <'a> arbitrary::Arbitrary<'a> {}
```